main/report.py

```diff
@@ -0,0 +1,280 @@
+        "description",
+    ]
+    queryset = models.MonthlyCharge.objects.filter(date=start_date).values(*fields)
+    charges_block = []
+    for record in queryset:
+        charges_block.append([str(record[field]) for field in fields])
+    return charges_block
+
+
+def get_csv_header_block(start_date: date) -> list[list[str]]:
+    """Get the header blocks for the CSV report.
+
+    Aggregates the total charge for the month across all monthly charges.
+
+    Args:
+        start_date: starting date (1st of the  month) for the report period
+
+    Returns:
+        A list of lists representing the 'header' rows in the CSV, excluding the rows
+            that include information on individual monthly charges
+    """
+    amount = models.MonthlyCharge.objects.filter(date=start_date).aggregate(
+        Sum("amount")
+    )["amount__sum"]
+    if amount:
+        amount = f"{amount:.2f}"
+
+    header_block = [
+        ["Journal Name", f"RCS_MANAGER RSE {start_date.strftime('%Y-%m')}", "", "", ""],
+        [
+            "Journal Description",
+            f"RCS RSE Recharge for {start_date.strftime('%Y-%m')}",
+            "",
+            "",
+            "",
+        ],
+        ["Journal Amount", str(amount), "", "", ""],
+        ["", "", "", "", ""],
+        ["Cost Centre", "Activity", "Analysis", "Credit", "Line Description"],
+        [
+            "ITPP",
+            "G80410",
+            "162104",
+            f"{amount}",
+            f"RSE Projects: {start_date.strftime('%B %Y')}",
+        ],
+        ["", "", "", "", ""],
+        ["Cost Centre", "Activity", "Analysis", "Debit", "Line Description"],
+    ]
+    return header_block
+
+
+def write_to_csv(
+    header_block: list[list[str]],
+    charges_block: list[list[str]],
+    writer: Writer,
+) -> None:
+    """Write CSV rows using CSV writer object.
+
+    Args:
+        header_block: list of lists representing rows in the CSV report for the header
+            blocks
+        charges_block: list of lists representing rows in the CSV report for the charges
+            block
+        writer: csv writer object
+    """
+    for block in [header_block, charges_block]:
+        for row in block:
+            writer.writerow(row)
+
+
+def create_charges_report(month: int, year: int, writer: Writer) -> None:
+    """Generate the CSV report by creating Monthly Charge objects and writing to a CSV.
+
+    Args:
+        month: month for the report date
+        year: year for the report date
+        writer: csv.writer to create the CSV report as HttpResponse or StringIO
+    """
+    # get the start_date and end dates (as the 1st of the month)
+    start_date = date(year, month, 1)
+    if start_date > datetime.today().date():
+        raise ValidationError("Report date must not be in the future.")
+
+    end_date_year = start_date.year + (start_date.month // 12)
+    end_date_month = (start_date.month % 12) + 1
+    end_date = date(end_date_year, end_date_month, 1)
```

Comment:
Maybe simpler:
```suggestion
    end_date = (start_date + timedelta(days=31)).replace(day=1)
```
This way, the change of month and year is taken care of automatically.